kronos-java/src/main/java/com/lyft/kronos/internal/ntp/SntpClient.java

```diff
@@ -119,6 +137,23 @@ public Response requestTime(String host, Long timeout) throws IOException {
         }
     }
 
+    private static void checkValidServerReply(
+            byte leap, byte mode, int stratum, long transmitTime)
+            throws IOException {
+        if (leap == NTP_LEAP_NOSYNC) {
+            throw new IOException("unsynchronized server");
```

Comment:
Can we create a custom exception for this?

Reply:
Instead of throwing IOException, u want to throw an InvalidServerResponseException (that extends IOException)?